front/app/containers/Admin/projects/project/files/components/FilesUpload/components/SelectedFile.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+
+import {
+  Box,
+  Icon,
+  Text,
+  Spinner,
+  colors,
+  Select,
+  Tooltip,
+} from '@citizenlab/cl2-component-library';
+import { CLError } from 'typings';
+
+import useAddFile from 'api/files/useAddFile';
+
+import Error from 'components/UI/Error';
+
+import { useIntl } from 'utils/cl-intl';
+import { getBase64FromFile } from 'utils/fileUtils';
+
+import { FileWithMeta } from '../types';
+
+import messages from './messages';
+
+type Props = {
+  fileMeta: FileWithMeta;
+  projectId: string;
+  onStatusUpdate: (status: Partial<FileWithMeta>) => void;
+};
+
+const SelectedFile = ({ fileMeta, projectId, onStatusUpdate }: Props) => {
+  const { formatMessage } = useIntl();
+  const { mutateAsync } = useAddFile();
+  const { file, status, semanticType } = fileMeta;
+  const [apiErrors, setApiErrors] = useState<CLError[] | null>(null);
```

Comment:
Is there to use React Hook Form for this instead of handling errors in the state? We try to avoid creating custom implementation where we can use established practices but maybe in this case it's not possible for some reason? Also not super critical, since it's a single field here but I was still curious.